eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -180,146 +195,156 @@ jobs:
-      fi
-    displayName: Build
-
-  # Don't run tests if overriding RID, we don't support that for now
-  - ${{ if eq(parameters.targetRid, '') }}:
     - script: |
         set -x
+        df -h
 
-        dockerVolumeArgs="-v $(sourcesPath):/vmr"
-        dockerEnvArgs="-e SMOKE_TESTS_EXCLUDE_OMNISHARP=${{ parameters.excludeOmniSharpTests }} -e SMOKE_TESTS_WARN_SDK_CONTENT_DIFFS=true -e SMOKE_TESTS_RUNNING_IN_CI=true"
-        poisonArg=''
+        # Allows Arcade to have access to the commit for the build
+        customEnvVars="BUILD_SOURCEVERSION=$BUILD_SOURCEVERSION"
+        customBuildArgs="--ci --prepareMachine"
+        if [[ '${{ parameters.runOnline }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --online"
+        fi
 
         if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
-          poisonArg='--poison'
+          customBuildArgs="$customBuildArgs --poison"
         fi
 
-        docker run --rm $dockerVolumeArgs -w /vmr $dockerEnvArgs ${{ parameters.container }} ./build.sh --source-only $poisonArg --test $(additionalBuildArgs) /p:SmokeTestConsoleVerbosity=detailed
-      displayName: Run Tests
-
-  # Don't use CopyFiles@2 as it encounters permissions issues because it indexes all files in the source directory graph.
-  - script: |
-      set -x
-
-      targetFolder=$(Build.StagingDirectory)/BuildLogs/
-      mkdir -p ${targetFolder}
-
-      cd "$(sourcesPath)"
-      find artifacts/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/prebuilt-report/ -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.diff" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.txt" -exec cp {} --parents -t ${targetFolder} \;
-    displayName: Prepare BuildLogs staging directory
-    continueOnError: true
-    condition: succeededOrFailed()
+        if [[ '${{ parameters.buildFromArchive }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-repository https://github.com/dotnet/dotnet"
+          customBuildArgs="$customBuildArgs --source-version $(git -C "${{ parameters.vmrPath }}" rev-parse HEAD)"
+        fi
+
+        if [[ '${{ parameters.buildSourceOnly }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-only"
+        fi
+
+        if [[ '${{ parameters.useMonoRuntime }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --use-mono-runtime"
+        fi
+
+        if [[ -n "${{ parameters.container }}" ]]; then
+          useDocker=true
```

Comment:
Is useDocker really necessary?  It is only referenced in one place.  It seems like parameters.container could be used and simplify the step a tiny bit.